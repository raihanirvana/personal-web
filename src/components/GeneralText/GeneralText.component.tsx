import { Props } from "./GeneralText.component.type";

const GeneralText = ({ customStyle, text }: Props) => (
  <p className={`text-white ${customStyle}`}>{text}</p>
);

export default GeneralText;
