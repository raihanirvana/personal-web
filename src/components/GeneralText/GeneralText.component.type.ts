export type Props = {
  text: string,
  customStyle: string,
}