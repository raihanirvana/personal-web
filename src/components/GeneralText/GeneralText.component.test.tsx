import { render } from "@testing-library/react";

import GeneralText from "./GeneralText.component";

describe('General Text', () => {
  it('must matches snapshot', () => {
    const { asFragment } = render(<GeneralText text="saya ganteng" customStyle="text-3xl" />)

    expect(asFragment()).toMatchSnapshot()
  })
})