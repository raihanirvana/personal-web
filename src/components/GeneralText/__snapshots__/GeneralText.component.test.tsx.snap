// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`General Text must matches snapshot 1`] = `
<DocumentFragment>
  <p
    class="text-white text-3xl"
  >
    saya ganteng
  </p>
</DocumentFragment>
`;
